File: Quick Left Rotation/Quick_Left_Rotation.cpp
class Solution {

  public:
    void leftRotate(vector<int>& arr, int k) {
        vector<int> temp = arr;
        int n = arr.size();
        k = k%n;
        if(k == 0) return;
        for(int i = 0; i<n; i++) {
            arr[i] = temp[(i+k)%n];
        }
        return;
    }
};

File: Quick Left Rotation/Quick_Left_Rotation.java
class Solution {
    void leftRotate(int arr[], int k) {
        int n = arr.length;
        k = k % n;
        if (k == 0) return;

        int[] temp = Arrays.copyOf(arr, n);

        for (int i = 0; i < n; i++) {
            arr[i] = temp[(i + k) % n];
        }
    }
}